### PR TITLE
 Update NIHMS schema for nlmta or (some issn with pubType)

### PR DESCRIPTION
**Pull request by birkland**
* NIHMS schema passes if a metadata blob contains an nlmta or _at least one_ issn contains a pubType.
  * This can be changed to _all_ issns contain a pubType by changing `contains` to `items` on line 31 of `schemas/jhu/nihms.json`. It is unclear which semantics match the assumptions being made by deposit services. 
* Added test schemas in `testdata/valid`, `testdata/invalid`, and enhanced validation test to verify expected passes or failures, due to the subtlety and complexity of this schema rule.

### CODE REVIEW

**Comment by emetsger** [comment]
@birkland right now the behavior of DS (specifically the [`NihmsMetadataSerializer`](https://github.com/OA-PASS/jhu-package-providers/blob/master/nihms-package-provider/src/main/java/org/dataconservancy/pass/deposit/provider/nihms/NihmsMetadataSerializer.java)) is:

* If an NLM-TA is present, include it in the metadata sent to PMC ([here](https://github.com/OA-PASS/jhu-package-providers/blob/master/nihms-package-provider/src/main/java/org/dataconservancy/pass/deposit/provider/nihms/NihmsMetadataSerializer.java#L103))
* If an ISSN is present, include it in the metadata sent to PMC ([here](https://github.com/OA-PASS/jhu-package-providers/blob/master/nihms-package-provider/src/main/java/org/dataconservancy/pass/deposit/provider/nihms/NihmsMetadataSerializer.java#L111))
    * ISSNs in Deposit Services are [modeled](https://github.com/OA-PASS/deposit-services/blob/master/deposit-model/src/main/java/org/dataconservancy/pass/deposit/model/DepositMetadata.java#L34) as a composite of the ISSN value itself and its publication type
    * Deposit Services assumes that if an ISSN is attached to a Journal, it has both the ISSN value and a publication type
    * My understanding is that NIH requires an ISSN with a pubtype when an ISSN is present in the metadata (i.e. an ISSN without a pubtype is not useful, and may invalidate the package sent to PMC)

Unless it's problematic, I'd recommend that we require both ISSN and publication type for _all_ ISSNs.  I'll let @jabrah weigh in, and see what he thinks.

**Comment by emetsger** [comment]
If the schema is changed to require all ISSNs have publication types, then Ember would be responsible for screening them prior to validation.   If the schema stays as-is, then Deposit Services should be updated to screen out any ISSNs with unknown publication types, including `null`.

**Comment by birkland** [comment]
It may also make sense for @jrmartino to weigh in as well.  Is the DOI service capable of creating a Journal record that does not have a pubType?  If not, then that might argue in favor of requiring publication type on all issns.

**Comment by jrmartino** [comment]
right now, the doi service will look for typed issns on the crossref record, and will keep track of their values. then it will look at untyped issns and see if there is anything not captured in the typed issn list. if so, it will add a typed issn with an empty type (":xxxx-xxxx").  this is to make sure that clients reading issns will always find the value at index 1 when the typed issn is split on ":"

**Comment by htpvu** [comment]
@emetsger @jabrah @birkland @jrmartino the DOI service does indeed create journal records with ISSN without pubType. We need to continue allowing this to happen because ISSN+pubtype combo requirement is specific to NIHMS deposit. Other repositories we know of (J10P and DASH) do not require this information. Blocking Journal creation due to the lack of ISSN would mean blocking any/all submissions if their journal iSSN lacks pubtype. The scope of this blockage would affect more than just NIHMS submissions. I hope that makes sense. 

In short, given the choices, we should update the Deposit Service to not send ISSN with unknown pubtypes. I'm counting on the high probability that most of the submission going to NIHMS will come from the pool of journals we collected from PMC list and MedLine database. These journals always have NLMTA. The possibility exists, however slim, that someone tries to submit an article in a journal unknown to NLM to NIHMS, but the best thing to do on that at this time, i think is to make sure it easy for us be aware of when that happens.

**Comment by birkland** [comment]
OK.  In that case, we can merge this as-is, and it'll just block submissions to NIHMS that neither have an nlmta, nor any issns that contain pubType.

**Comment by emetsger** [comment]
@jrmartino @htpvu So this means that Deposit services needs to be able to parse an ISSN that looks like: `:xxx-xxx-xxx-xxx`?
